include/moveit_grasps/suction_grasp_generator.h

```diff
@@ -0,0 +1,122 @@
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  /**
+   * \brief Constructor
+   */
+  SuctionGraspGenerator(moveit_visual_tools::MoveItVisualToolsPtr visual_tools, bool verbose = false);
+
+  /**
+   * \brief Create possible grasp positions around a cuboid
+   * \param cuboid_pose - centroid of object to grasp in world frame
+   * \param depth length of cuboid along local x-axis
+   * \param width length of cuboid along local y-axis
+   * \param height length of cuboid along local z-axis
+   * \param grasp_data data describing end effector
+   * \param grasp_candidate_config parameter for selectively enabling and disabling different grasp types
+   * \param grasp_candidates possible grasps generated
+   * \return true if successful
+   */
+  bool generateGrasps(const Eigen::Isometry3d& cuboid_pose, double depth, double width, double height,
+                      const SuctionGraspDataPtr& grasp_data, std::vector<GraspCandidatePtr>& grasp_candidates);
+
+  /**
+   * \brief creates grasp messages from the generated grasp poses
+   * \param grasp_pose_eef_mount - the grasp pose. (Note: this is the pose of the eef mount not the position of the tcp)
+   * \param grasp_data data describing the end effector
+   * \param grasp_candidates - list possible grasps
+   * \param object_pose - pose of object to grasp
+   * \param object_size - size of object to grasp
+   * \param object_width - In the case of finger grippers, the width of the object in the dimension betwen the fingers
+   * \return true on success
+   */
+  bool addGrasp(const Eigen::Isometry3d& grasp_pose_eef_mount, const SuctionGraspDataPtr& grasp_data,
+                std::vector<GraspCandidatePtr>& grasp_candidates, const Eigen::Isometry3d& object_pose,
+                const Eigen::Vector3d& object_size, double object_width);
+
+  /**
+   * \brief Score the generated suction grasp poses
+   * \param grasp_pose_tcp - the pose of the grasp
+   * \param grasp_data - data describing the end effector
+   * \param cuboid_pose - the pose of the object being grasped
+   * \param object size - the extents of the object being grasped
+   * \param suction_voxel_overlap - all voxels with a percentage of coverage above some cutoff
+   * \return a score with positive being better
+   */
+  double scoreSuctionGrasp(const Eigen::Isometry3d& grasp_pose_tcp, const SuctionGraspDataPtr& grasp_data,
+                           const Eigen::Isometry3d& cuboid_pose, const Eigen::Vector3d& object_size,
+                           std::vector<double>& suction_voxel_overlap);
+
+protected:
+  bool generateSuctionGrasps(const Eigen::Isometry3d& cuboid_top_pose, double depth, double width, double height,
+                             const SuctionGraspDataPtr& grasp_data, std::vector<GraspCandidatePtr>& grasp_candidates);
+
+public:
+  // Visual debug settings
+  bool debug_top_grasps_;
+  bool show_grasp_overhang_;
+
+  SuctionGraspScoreWeights grasp_score_weights_;
```

Comment:
seemed cleaner than adding setter and getter for it. Is this worse?

Reply:
yes because the _ at the end of the variable comes from the google code standard and implies it is a private variable.  Instead of having these as public variables they should just be function parameters to the functions that need them.  Make the caller retain that state in it's own scope.

Reply:
Ok. That makes sense. I agree this would be better and I don't like how everything in MG is public but I don't think that fixing it is in scope for this refactor